### PR TITLE
cleanup in dash testcases 

### PR DESCRIPTION
**Pull request by vincent-xs**
addressing a few minor issues encountered and fix locally while running testcases and developing SAI.

- dash testcase cleanup:
return proper objects
fix remove command generation in scale outbound (we need the whole key in the remove command)

- update saithrift docker file (-qq is not working on newer docker compose version)

- update p4 generation to handle ip prefix list (needed when we compile dash acl with DASH_MATCH to generate the proper SAI API)

### CODE REVIEW

**Comment by KrisNey-MSFT** [comment]
Perhaps place some GitHub checkboxes to split this up, etc...

**Review by chrispsommers — APPROVED** [review]
LGTM - needs review from someone at Intel to evaluate potential impact. @reshmaintel to assign someone

**Comment by yharhasx** [comment]
Hi @vincent-xs, thanks for your fixes, they look good to me!
But I have few suggestions regarding Metering API in sai_dash_utils.py:
- As it`s cleanup PR I believe that would be better to introduce metering API changes is separate PR.
- Moreover, we are planning to create a PR with adding of Metering API, CRUD and VNET to VNET test cases in the end of the next week after final run on real HW.

So, to sum up, would it be suitable for you to update your PR without Metering changes? And in case of any suggestions from your side create new one based on our future Metering PR.

**Comment by vincent-xs** [comment]
> Hi @vincent-xs, thanks for your fixes, they look good to me! But I have few suggestions regarding Metering API in sai_dash_utils.py:
> 
> * As it`s cleanup PR I believe that would be better to introduce metering API changes is separate PR.
> * Moreover, we are planning to create a PR with adding of Metering API, CRUD and VNET to VNET test cases in the end of the next week after final run on real HW.
> 
> So, to sum up, would it be suitable for you to update your PR without Metering changes? And in case of any suggestions from your side create new one based on our future Metering PR.

Yuri,

wrt metering, I can certainly remove those changes (which I used to UT my code change). I got to believe similar changes will show up in the PR you'll be creating to support those new testcases. once your PR gets committed, I'll figure out if those changes even need to be there.

**Comment by yharhasx** [comment]
> > Hi @vincent-xs, thanks for your fixes, they look good to me! But I have few suggestions regarding Metering API in sai_dash_utils.py:
> > 
> > * As it`s cleanup PR I believe that would be better to introduce metering API changes is separate PR.
> > * Moreover, we are planning to create a PR with adding of Metering API, CRUD and VNET to VNET test cases in the end of the next week after final run on real HW.
> > 
> > So, to sum up, would it be suitable for you to update your PR without Metering changes? And in case of any suggestions from your side create new one based on our future Metering PR.
> 
> Yuri,
> 
> wrt metering, I can certainly remove those changes (which I used to UT my code change). I got to believe similar changes will show up in the PR you'll be creating to support those new testcases. once your PR gets committed, I'll figure out if those changes even need to be there.

Vincent, 
Thanks for understanding! Our code related to metering has changes similar to yours and easily can be used for new test cases development etc.
Thanks again!

**Review by yharhasx — CHANGES_REQUESTED** [review]

File: test/test-cases/functional/ptf/sai_dash_utils.py [hunk 1]
```diff
@@ -341,11 +351,13 @@ def outbound_ca_to_pa_create(self, dst_vnet_id, dip, underlay_dip,
         sai_thrift_create_outbound_ca_to_pa_entry(self.client, ca_to_pa_entry,
                                                   underlay_dip=sai_ipaddress(underlay_dip),
                                                   use_dst_vnet_vni=use_dst_vnet_vni,
-                                                  overlay_dmac=overlay_dmac,
+                                                  overlay_dmac=overlay_dmac0,
```
[the remark this answers is not in the document]
Comment (vincent-xs) [reply test/test-cases/functional/ptf/sai_dash_utils.py:354]: indeed, I didn't have that in my original diff.

**Pull request merged**